### ISSUE TITLE
BepInEx starts, then closes and restarts the game

### ISSUE THREAD

**Issue by RandomWolf** [BepInEx/BepInEx#626]
### What happened?

**What happened?**
I have installed the latest BepInEx 5 version from this GitHub repository and installed in the new demo release of Wizard with a Gun.  
I have reconfigured BepInEx to show the console with the log level set to All  
Upon starting the game via Steam, the game starts without the BepInEx Console  
When starting the game via the executable in the folder, however, the BepInEx Console opens, goes through the Preloading sequence, but after everything is done the console window closes and tries to relaunch the game via steam, which is unsuccessfull since the game was already running "in a sense".

### BepInEx Distribution

Stable from GitHub

### Log outputs

```txt
[Message:   BepInEx] BepInEx 5.4.21.0 - wizardwithagun (09.06.2023 00:27:16)
[Info   :   BepInEx] Running under Unity v2020.3.37.9201280
[Info   :   BepInEx] CLR runtime version: 4.0.30319.42000
[Info   :   BepInEx] Supports SRE: True
[Info   :   BepInEx] System platform: Bits64, Windows
[Message:   BepInEx] Preloader started
[Info   :   BepInEx] Loaded 1 patcher method from [BepInEx.Preloader 5.4.21.0]
[Info   :   BepInEx] Loaded 1 patcher method from [BepInEx.MonoMod.HookGenPatcher 1.2.1.0]
[Info   :   BepInEx] Loaded 1 patcher method from [StartupProfiler 10.0.0.0]
[Info   :   BepInEx] 3 patcher plugins loaded
[Info   :   Console] [MonoMod] 
[Info   :   Console] Reading input file into module.
[Info   :HookGenPatcher] Starting HookGenerator
[Info   :HookGenPatcher] Done.
[Info   :   BepInEx] Patching [UnityEngine.CoreModule] with [BepInEx.Chainloader]
[Message:   BepInEx] Preloader finished
[Message:   BepInEx] Chainloader ready
[Message:   BepInEx] Chainloader started
[Info   :   BepInEx] 0 plugins to load
[Message:   BepInEx] Chainloader startup complete
```

### Environment

```markdown
- OS: Windows 10
- BepInEx: 5.4.21
- Game: Wizard With A Gun Demo
```

**Comment by mainpsyhos** [comment]
### Same issue.

Upon starting the game via Steam, the game starts without the BepInEx Console.
Upon direct starting, the BepInEx Console opens, goes through the Preloading sequence, then game restarts via steam **without** console. 

### Log outputs
```
[Message:   BepInEx] BepInEx 5.4.22.0 - Timberborn (29.10.2023 20:03:02)
[Info   :   BepInEx] Running under Unity v2022.2.17.5557147
[Info   :   BepInEx] CLR runtime version: 4.0.30319.42000
[Info   :   BepInEx] Supports SRE: True
[Info   :   BepInEx] System platform: Bits64, Windows
[Message:   BepInEx] Preloader started
[Info   :   BepInEx] Loaded 1 patcher method from [BepInEx.Preloader 5.4.22.0]
[Info   :   BepInEx] 1 patcher plugin loaded
[Info   :   BepInEx] Patching [UnityEngine.CoreModule] with [BepInEx.Chainloader]
[Message:   BepInEx] Preloader finished
[Message:   BepInEx] Chainloader ready
[Message:   BepInEx] Chainloader started
[Info   :   BepInEx] 1 plugin to load
[Info   :   BepInEx] Loading [Mod Manager UI 1.2.4]
[Message:   BepInEx] Chainloader startup complete
[Info   : Unity Log] Starting game version 0.4.9.3-6c7fb02-sw
```

### Environment
```
- OS: Windows 10
- BepInEx: 5.4.22 (actual)
- Game: Timberborn, both stable (update 4) and experimental (update 5) branches
```

**Comment by cracktorio** [comment]
i am also having this issue, with the latest version of subnautica. This didn't happen when installed normally, only after i copied it to another folder so that i could have the legacy version without having to keep redownloading. After exiting it will open subnautica via steam, giving me the unmodded legacy version.

**Comment by mamijomale** [comment]
Alright, I had a similar issue while trying to mod the game Armello, and I actually found a way to make our beloved console to not vanish (so I hope it's the same for all other game with this issue) : 

**put the modified winhttp.dll in Steam folder, where Steam.exe is located.**

Here are the steps I followed to find this solution : 

  -> We have to launch the game via Steam as launching the game via the exe in the game folder would always lead to closing itself and launching via Steam anyway.

 -> What I've noticed : if the game is launched via the game folder it results in seeing the console until loaded and closed then reopened via Steam. And if the game is launched via Steam directly, the console never shows up. I thus started to have suspicion on "winhttp.dll" being somehow ignored by steam.

 -> Considering winhttp is searched nearby the exe file first, then the sys32 folder, I thought Steam was not looking for winhttp.dll in the gamefolder but either directly from sys32 (which would have been silly dark magic) or elsewhere.

 -> Thinking about the fact we're using Steam.exe to launch our game exe, I had the silly theory : what if the dll wasn't just search from the location of Steam.exe then sys32 instead? So I've put winhttp.exe in Steam folder, were Steam.exe is located.

RESULT : SUCCESS
Now my Armello game keeps the console active like I wanted and I can see the log at the right position, and cherry on the top : the plugin loads normally and doesn't seem to mix up with other game on which I used BepInEx.

My guess is Steam only load winhttp next to Steam.exe, but does all the rest from the game folder. Maybe it's all about priority? I don't know but well, at least, It worked for me

That's all!
I wish you all of you for the best and hope that solution will work for you as well!
